### DISCUSSION TITLE
Using css custom properties for colors in combination with background opacity does not work

### DISCUSSION THREAD

**Discussion by kolaente** [tailwindlabs/tailwindcss#7407] [Help]
Hi!

We have custom colors defined with css variables. The tailwind config contains something like this for the color:

```js
module.exports = {
  theme: {
    extend: {
      colors: {
        primary: 'var(--primary)'
      }
    },
  },
}
```

In the css, we then set the `--primary` property to the color we want to use.

This is similar to the process shown in [this video](https://www.youtube.com/watch?v=MAtaT8BZEAo).

Setting the color works great, but it does not look like it's possible to use this in combination with a background opacity.
Tailwind seems to always generate this CSS:

```css
.bg-primary {
  background-color: var(--primary);
}
```

vs a "real" color generates this CSS:

```css
.bg-red-400 {
  --tw-bg-opacity: 1;
  background-color: rgb(248 113 113 / var(--tw-bg-opacity));
}
```

The `--tw-bg-opacity` property seems to allow setting the background opacity (which is a really nice way of doing that, props to the team for making this work!)

This actually works with css as well, but only if the color is in the format the `rgba` function expects and I guess tailwind has no way of figuring out what color format actually is behind the variable?

Fiddle: https://jsfiddle.net/x0hn56p2/1/

Is there a way to get this to work? One way would of course be to redefine all `bg-` classes in css so they include the color and the `--tw-bg-opacity` property but that feels very dirty. Fixing this would probably require changing the parser.

We also had the idea to change the color definition in the tailwind config file to something like `rgba(var(--primary) / var(--tw-bg-opacity))` but that would also affect all other uses of the color when not used as a background color.

Minimal example: https://play.tailwindcss.com/Lzn4isgHO9

**Comment by jreviews** [comment]
Does this help https://tailwindcss.com/docs/customizing-colors#using-css-variables?
